packages/terra-disclosure-manager/tests/jest/DisclosureManager.test.jsx

```diff
@@ -0,0 +1,521 @@
+    );
+
+    const wrapper = shallow(disclosureManager);
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  const mountDisclosureManager = (diclosureTypes, renderFunc) => shallow((
+    <DisclosureManager
+      render={renderFunc}
+      supportedDisclosureTypes={diclosureTypes}
+    >
+      <div id="child1" />
+      <div id="child2" />
+    </DisclosureManager>
+    ));
+
+  const validateInitialState = (wrapper) => {
+    expect(wrapper.state().disclosureIsOpen).toBeFalsy();
+    expect(wrapper.state().disclosureIsFocused).toBeTruthy();
+    expect(wrapper.state().disclosureIsMaximized).toBeFalsy();
+    expect(wrapper.state().disclosureSize).toBe('small');
+    expect(wrapper.state().disclosureComponentKeys).toEqual([]);
+    expect(wrapper.state().disclosureComponentData).toEqual({});
+  };
+
+  const validateChildAppDelegate = (wrapper) => {
+    const childApp1 = wrapper.find('#child1').getElements()[0].props.app;
+    expect(childApp1).toBeDefined();
+    expect(childApp1.disclose).toBeDefined();
+
+    const childApp2 = wrapper.find('#child2').getElements()[0].props.app;
+    expect(childApp2).toBeDefined();
+    expect(childApp2.disclose).toBeDefined();
+  };
+
+  const triggerChildDisclose = wrapper => (
+    new Promise((resolve, reject) => {
+      const childApp1 = wrapper.find('#child1').getElements()[0].props.app;
+      childApp1.disclose({
+        preferredType: 'test',
+        size: 'large',
+        content: {
+          key: 'DISCLOSE_KEY',
+          component: <div id="disclosure-component" />,
+        },
+      }).then(resolve).catch(reject);
+    })
+    .then(() => {
+      wrapper.update();
+
+      expect(wrapper.state().disclosureIsOpen).toBeTruthy();
+      expect(wrapper.state().disclosureIsFocused).toBeTruthy();
+      expect(wrapper.state().disclosureIsMaximized).toBeFalsy();
+      expect(wrapper.state().disclosureSize).toBe('large');
+      expect(wrapper.state().disclosureComponentKeys).toEqual(['DISCLOSE_KEY']);
+      expect(wrapper.state().disclosureComponentData.DISCLOSE_KEY.key).toEqual('DISCLOSE_KEY');
+      expect(wrapper.state().disclosureComponentData.DISCLOSE_KEY.component.props.id).toEqual('disclosure-component');
+
+      const disclosureContentApp = wrapper.find('#disclosure-component').getElements()[0].props.app;
+      expect(disclosureContentApp).toBeDefined();
+      expect(disclosureContentApp.disclose).toBeDefined();
+      expect(disclosureContentApp.dismiss).toBeDefined();
+      expect(disclosureContentApp.closeDisclosure).toBeDefined();
+      expect(disclosureContentApp.goBack).toBeUndefined();
+      expect(disclosureContentApp.maximize).toBeDefined();
+      expect(disclosureContentApp.minimize).toBeUndefined();
+      expect(disclosureContentApp.requestFocus).toBeDefined();
+      expect(disclosureContentApp.releaseFocus).toBeUndefined();
+      expect(disclosureContentApp.registerDismissCheck).toBeDefined(); // TODO: Flip this with AppDelegate release
```

Comment:
Should this todo be completed or has this been completed already?

Reply:
It's been done now, I'll remove the comments.